src/multi_physics/segregated_fsi_solver.cc

```diff
@@ -942,11 +958,19 @@ namespace oomph
           // Throw an error indicating if we ran out of iterations
           if (iter_taken == Max_picard)
           {
-            throw SegregatedSolverError(true);
+            throw SegregatedSolverError(
+              "Error occured in Segregated solver. \n",
+              OOMPH_CURRENT_FUNCTION,
+              OOMPH_EXCEPTION_LOCATION,
+              true);
           }
           else
           {
-            throw SegregatedSolverError(false);
+            throw SegregatedSolverError(
+              "Error occured in Segregated solver. \n",
+              OOMPH_CURRENT_FUNCTION,
+              OOMPH_EXCEPTION_LOCATION,
+              false);
```

Comment:
I don't like the flag in the error arguments. I think we should throw two different errors instead.